faulthandler.c

```diff
@@ -151,6 +151,29 @@ extern const char* _Py_DumpTracebackThreads(
     PyInterpreterState *interp,
     PyThreadState *current_thread);
 
+
+#define TIMEBUFF_LEN 21 // enough space to represent a "long long" time
+static char _timebuff[TIMEBUFF_LEN] = {0};
```

Comment:
Don't use static buffer, but allocate memory on the stack, to handle correctly reentrant calls. Signals can be evil!